File: src/plugin/rsinterface/ChatSetupInterfacePlugin.java
package plugin.rsinterface;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class ChatSetupInterfacePlugin implements InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		if (componentId == 5) {
			player.getInterfaceManager().sendSettings();
		} else if (componentId == 42) {
			player.getAttributes().setPrivateChatSetup(player.getAttributes().getPrivateChatSetup() == 0 ? 1 : 0);
		} else if (componentId >= 49 && componentId <= 61) {
			player.getAttributes().setPrivateChatSetup(componentId - 48);
		}
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(982);
	}
}

File: src/main/java/plugin/rsinterface/ChatSetupInterfacePlugin.kt
package plugin.rsinterface

import game.content.plugin.type.InterfacePlugin
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 8/31/2017
 */
class ChatSetupInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int
    ): Boolean {
        when (componentId) {
            5 -> {
                player.interfaceManager.sendSettings()
            }
            42 -> {
                player.attributes.privateChatSetup = if (player.attributes.privateChatSetup == 0) 1 else 0
            }
            in 49..61 -> {
                player.attributes.privateChatSetup = componentId - 48
            }
        }
        return true
    }

    override fun register() {
        registerInterfacePlugin(982)
    }
}